aks-node-controller/app.go

```diff
@@ -77,8 +77,12 @@ func (a *App) run(ctx context.Context, args []string) error {
 		provisionOutput, err := a.ProvisionWait(ctx, provisionStatusFiles)
 		//nolint:forbidigo // stdout is part of the interface
 		fmt.Println(provisionOutput)
+		if err != nil {
+			slog.Error("aks-node-controller failed", "error", err)
```

Comment:
Usually, there is a no reason to log error and return it at the same time.

The same error will be logged multiple times and you will get a lot of noise.
I usually log errors only once when discarding it. If any extra context is needed, add it with `fmt.Errorf`

Reply:
It sounds like error logging in `App.Run()` isn't right?
And in some edge-cases an error is silently ignored?